<a href="https://colab.research.google.com/github/Natalys2003/Project2021/blob/main/%D0%9A%D0%BE%D0%BF%D0%B8%D1%8F_%D0%B1%D0%BB%D0%BE%D0%BA%D0%BD%D0%BE%D1%82%D0%B0_%22%D0%97%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5_1_ipynb%22.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Задание по практике 1

##Загрузка датасета
В файле classification datasets расположены ссылки на датасеты. Выберите один в соответствии со своим вариантом, скачайте и прочитайте файл

**Не забудьте загрузить нужные библиотеки**

Например: pip3 install pandas

In [ ]:
import pandas as pd

In [ ]:
df =pd.read_csv("/content/sample_data/california_housing_test.csv")
df

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0       -122.05     37.37                27.0       3885.0           661.0   
1       -118.30     34.26                43.0       1510.0           310.0   
2       -117.81     33.78                27.0       3589.0           507.0   
3       -118.36     33.82                28.0         67.0            15.0   
4       -119.67     36.33                19.0       1241.0           244.0   
...         ...       ...                 ...          ...             ...   
2995    -119.86     34.42                23.0       1450.0           642.0   
2996    -118.14     34.06                27.0       5257.0          1082.0   
2997    -119.70     36.30                10.0        956.0           201.0   
2998    -117.12     34.10                40.0         96.0            14.0   
2999    -119.63     34.42                42.0       1765.0           263.0   

      population  households  median_income  median_house_value  
0         1537.0       606.0         6.6085            344700.0  
1          809.0       277.0         3.5990            176500.0  
2         1484.0       495.0         5.7934            270500.0  
3           49.0        11.0         6.1359            330000.0  
4          850.0       237.0         2.9375             81700.0  
...          ...         ...            ...                 ...  
2995      1258.0       607.0         1.1790            225000.0  
2996      3496.0      1036.0         3.3906            237200.0  
2997       693.0       220.0         2.2895             62000.0  
2998        46.0        14.0         3.2708            162500.0  
2999       753.0       260.0         8.5608            500001.0  

[3000 rows x 9 columns]

##Первое знакомство с датасетом
В следующих ячейках выведите:


1.   Колонки датасета
2.   Первые 5 значений
3.   Последние 5 значений
4.   Статистику по числовым колонкам



In [ ]:
df.columns

Index(['longitude', 'latitude', 'housing_median_age', 'total_rooms',
       'total_bedrooms', 'population', 'households', 'median_income',
       'median_house_value'],
      dtype='object')

In [ ]:
#-122.05, -118.30, -117.81, -118.36, -119.67

In [ ]:
#344700.0, 176500.0, 270500.0, 330000.0, 81700.0

In [ ]:
df.describe()

longitude    latitude  housing_median_age   total_rooms  \
count  3000.000000  3000.00000         3000.000000   3000.000000   
mean   -119.589200    35.63539           28.845333   2599.578667   
std       1.994936     2.12967           12.555396   2155.593332   
min    -124.180000    32.56000            1.000000      6.000000   
25%    -121.810000    33.93000           18.000000   1401.000000   
50%    -118.485000    34.27000           29.000000   2106.000000   
75%    -118.020000    37.69000           37.000000   3129.000000   
max    -114.490000    41.92000           52.000000  30450.000000   

       total_bedrooms    population  households  median_income  \
count     3000.000000   3000.000000  3000.00000    3000.000000   
mean       529.950667   1402.798667   489.91200       3.807272   
std        415.654368   1030.543012   365.42271       1.854512   
min          2.000000      5.000000     2.00000       0.499900   
25%        291.000000    780.000000   273.00000       2.544000   
50%        437.000000   1155.000000   409.50000       3.487150   
75%        636.000000   1742.750000   597.25000       4.656475   
max       5419.000000  11935.000000  4930.00000      15.000100   

       median_house_value  
count          3000.00000  
mean         205846.27500  
std          113119.68747  
min           22500.00000  
25%          121200.00000  
50%          177650.00000  
75%          263975.00000  
max          500001.00000

**Ответьте на следующие вопросы:**


1.   Какие средние значения в каждой числовой колонке вашего датасета?
2.   Минимальные и максимальные значения в каждой числовой колонке вашего датасета





1.   **Ответ 1.** 1.994936,	2.12967,	12.555396,	2155.593332,	415.654368,	1030.543012,	365.42271,	1.854512,	113119.68747
2.   **Ответ 2.** min: -124.180000,	32.56000,	1.000000,	6.000000,	2.000000,	5.000000,	2.00000,	0.499900,	22500.00000
max:	-114.490000,	41.92000,	52.000000,	30450.000000,	5419.000000,	11935.000000,	4930.00000,	15.000100,	500001.00000


##Типизация
Присвойте соответствующим колонкам один из типов (category, float, int)

In [ ]:
df.longitude = df.longitude.astype("category")
df.info()
#Вывести типы колонок тут

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 9 columns):
 #   Column              Non-Null Count  Dtype   
---  ------              --------------  -----   
 0   longitude           3000 non-null   category
 1   latitude            3000 non-null   float64 
 2   housing_median_age  3000 non-null   float64 
 3   total_rooms         3000 non-null   float64 
 4   total_bedrooms      3000 non-null   float64 
 5   population          3000 non-null   float64 
 6   households          3000 non-null   float64 
 7   median_income       3000 non-null   float64 
 8   median_house_value  3000 non-null   float64 
dtypes: category(1), float64(8)
memory usage: 214.4 KB


Какие признаки вы можете выделить в своем датасете (категориальные, числовые, числовые количественные)

##Фильтрация
Офильтруйте 3 колонки вашего датасета и выведите результат

In [ ]:
df[df["latitude"] > 13]

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0      -122.05     37.37                27.0       3885.0           661.0   
1      -118.30     34.26                43.0       1510.0           310.0   
2      -117.81     33.78                27.0       3589.0           507.0   
3      -118.36     33.82                28.0         67.0            15.0   
4      -119.67     36.33                19.0       1241.0           244.0   
...        ...       ...                 ...          ...             ...   
2995   -119.86     34.42                23.0       1450.0           642.0   
2996   -118.14     34.06                27.0       5257.0          1082.0   
2997   -119.70     36.30                10.0        956.0           201.0   
2998   -117.12     34.10                40.0         96.0            14.0   
2999   -119.63     34.42                42.0       1765.0           263.0   

      population  households  median_income  median_house_value  
0         1537.0       606.0         6.6085            344700.0  
1          809.0       277.0         3.5990            176500.0  
2         1484.0       495.0         5.7934            270500.0  
3           49.0        11.0         6.1359            330000.0  
4          850.0       237.0         2.9375             81700.0  
...          ...         ...            ...                 ...  
2995      1258.0       607.0         1.1790            225000.0  
2996      3496.0      1036.0         3.3906            237200.0  
2997       693.0       220.0         2.2895             62000.0  
2998        46.0        14.0         3.2708            162500.0  
2999       753.0       260.0         8.5608            500001.0  

[3000 rows x 9 columns]

In [ ]:
df[df["housing_median_age"] < 13]

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
33     -118.08     34.55                 5.0      16181.0          2971.0   
39     -119.01     34.23                11.0       5785.0          1035.0   
45     -117.24     33.17                 4.0       9998.0          1874.0   
48     -121.35     38.00                 6.0       1649.0           369.0   
56     -121.42     38.72                10.0       3054.0           528.0   
...        ...       ...                 ...          ...             ...   
2969   -118.11     34.68                 6.0       7430.0          1184.0   
2971   -120.93     35.76                11.0       8997.0          1698.0   
2975   -120.71     35.50                12.0       3098.0           453.0   
2977   -119.70     36.75                11.0       3626.0           779.0   
2997   -119.70     36.30                10.0        956.0           201.0   

      population  households  median_income  median_house_value  
33        8152.0      2651.0         4.5237            141800.0  
39        2760.0       985.0         4.6930            232200.0  
45        3925.0      1672.0         4.2826            237500.0  
48         732.0       350.0         3.4231            123800.0  
56        1932.0       510.0         3.0903             91900.0  
...          ...         ...            ...                 ...  
2969      3489.0      1115.0         5.3267            140100.0  
2971      1825.0       756.0         3.2300            154300.0  
2975      1433.0       434.0         5.2508            292900.0  
2977      1819.0       731.0         2.4956             87500.0  
2997       693.0       220.0         2.2895             62000.0  

[296 rows x 9 columns]

In [ ]:
df[df["population"] > 1003]

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0      -122.05     37.37                27.0       3885.0           661.0   
2      -117.81     33.78                27.0       3589.0           507.0   
7      -120.65     35.48                19.0       2310.0           471.0   
8      -122.84     38.40                15.0       3080.0           617.0   
9      -118.02     34.08                31.0       2402.0           632.0   
...        ...       ...                 ...          ...             ...   
2988   -122.01     36.97                43.0       2162.0           509.0   
2989   -122.02     37.60                32.0       1295.0           295.0   
2990   -118.23     34.09                49.0       1638.0           456.0   
2995   -119.86     34.42                23.0       1450.0           642.0   
2996   -118.14     34.06                27.0       5257.0          1082.0   

      population  households  median_income  median_house_value  
0         1537.0       606.0         6.6085            344700.0  
2         1484.0       495.0         5.7934            270500.0  
7         1341.0       441.0         3.2250            166900.0  
8         1446.0       599.0         3.6696            194400.0  
9         2830.0       603.0         2.3333            164200.0  
...          ...         ...            ...                 ...  
2988      1208.0       464.0         2.5417            260900.0  
2989      1097.0       328.0         3.2386            149600.0  
2990      1500.0       430.0         2.6923            150000.0  
2995      1258.0       607.0         1.1790            225000.0  
2996      3496.0      1036.0         3.3906            237200.0  

[1793 rows x 9 columns]

##Группировка
Сгруппируйте датасет по какому-либо категориальному признаку и выведите среднее значение численных колонок (см. метод mean в pandas)

In [ ]:

df.groupby(["longitude"]).count()

latitude  housing_median_age  total_rooms  total_bedrooms  \
longitude                                                              
-124.18           1                   1            1               1   
-124.17           1                   1            1               1   
-124.16           4                   4            4               4   
-124.15           1                   1            1               1   
-124.14           3                   3            3               3   
...             ...                 ...          ...             ...   
-114.98           1                   1            1               1   
-114.62           1                   1            1               1   
-114.61           1                   1            1               1   
-114.55           1                   1            1               1   
-114.49           1                   1            1               1   

           population  households  median_income  median_house_value  
longitude                                                             
-124.18             1           1              1                   1  
-124.17             1           1              1                   1  
-124.16             4           4              4                   4  
-124.15             1           1              1                   1  
-124.14             3           3              3                   3  
...               ...         ...            ...                 ...  
-114.98             1           1              1                   1  
-114.62             1           1              1                   1  
-114.61             1           1              1                   1  
-114.55             1           1              1                   1  
-114.49             1           1              1                   1  

[607 rows x 8 columns]

##Задание для гениев *
Агрегируйте числовой признак по категориальному признаку с использованием метода **без использования** библиотеки pandas, и вычислите медиану и среднее значение для каждой категории.

In [ ]:
#Агрегация